spec/controllers/profiles_controller_spec.rb

```diff
@@ -3,306 +3,304 @@
+          delete :destroy, params: { id: ada.id }
+          expect(response).to redirect_to("/#{I18n.locale}/profiles")
+        end
       end
 
-      it 'should redirect to profiles overview' do
-        delete :destroy, params: { id: ada.id }
-        expect(response).to redirect_to("/#{I18n.locale}/profiles")
-      end
-    end
+      context 'when trying to destroy a different profile' do
+        before do
+          sign_in ada
+          delete :destroy, params: { id: profile_published.id }
+        end
 
-    context 'when trying to destroy a different profile' do
-      before do
-        sign_in ada
-        delete :destroy, params: { id: profile_published.id }
-      end
+        it 'should not destroy the requested profile' do
+          expect(Profile.where(id: profile_published.id).count).to eq 1
+        end
 
-      it 'should not destroy the requested profile' do
-        expect(Profile.where(id: profile_published.id).count).to eq 1
+        it 'should redirect to profiles overview' do
+          expect(response).to redirect_to("/#{I18n.locale}/profiles")
+        end
       end
 
-      it 'should redirect to profiles overview' do
-        expect(response).to redirect_to("/#{I18n.locale}/profiles")
+      context 'when trying destroy profile if user is not signed in' do
+        before do
+          delete :destroy, params: { id: ada.id }
+        end
+
+        it 'should not destroy the requested profile' do
+          expect(Profile.where(id: ada.id).count).to eq 1
+        end
+
+        it 'should redirect to profiles overview' do
+          expect(response).to redirect_to("/#{I18n.locale}/profiles")
+        end
       end
     end
+  end
 
-    context 'when trying destroy profile if user is not signed in' do
-      before do
-        delete :destroy, params: { id: ada.id }
-      end
+  context "pagination with multiple profiles" do
+    it 'displays no profiles on index page 2' do
+      published_profiles = create_list(:published_profile, 24, main_topic_en: 'math')
+      get :index, params: { page: 1 }
+      expect(assigns(:records).count).to eq 24
+      expect(assigns(:records)).to eq published_profiles
+      get :index, params: { page: 2 }
+      expect(assigns(:records).count).to eq 0
+    end
 
-      it 'should not destroy the requested profile' do
-        expect(Profile.where(id: ada.id).count).to eq 1
-      end
+    it 'displays no profiles twice on index page 2' do
+      create_list(:published_profile, 26, main_topic_en: 'math')
 
-      it 'should redirect to profiles overview' do
-        expect(response).to redirect_to("/#{I18n.locale}/profiles")
-      end
+      get :index, params: { page: 1 }
+      expect(assigns(:records).count).to eq 24
+      profiles_page_1 = assigns(:records)
+
+      get :index, params: { page: 2 }
+      expect(assigns(:records).count).to eq 2
+      profiles_page_2 = assigns(:records)
+      expect(profiles_page_2 & profiles_page_1).to eq []
```

Comment:
Sieht elegant aus, aber eigentlich verstehe ich nicht genau, was & hier macht. Ist das sowas wie eine Schnittmenge?

Reply:
jep it checks for an intersection. I wanted to check that we shall no profile twice.